### PR TITLE
feat(AutoSuggestField Menu): Fixes debounce issue and menu

### PR DESCRIPTION
**Pull request by yasabere**
affects: @rentpath/react-ui-core

[Card](https://rentpath.leankit.com/card/634890555)
- Added ability to include non-clickable options in suggestions
- Added tests
- Options are disabled by adding 'disabled:true' attribute to option object

### CODE REVIEW

File: packages/react-ui-core/src/Menu/Menu.js [hunk 1]
```diff
@@ -93,28 +95,26 @@ export default class Menu extends PureComponent {
...
-
   get options() {
     return this.props.options || []
   }
 
   get items() {
-    return this.options.map(option => (
-      typeof option === 'object' ? option.label : option
-    ))
+    return this.options.map(option => ((typeof option === 'object' && option.label !== undefined) ? option.label : option))
```
Comment (mneumark) [inline packages/react-ui-core/src/Menu/Menu.js]: I think this would be more readable if the ternary were on a separate line like it was before.

File: packages/react-ui-core/src/Menu/Menu.js [hunk 2]
```diff
@@ -46,6 +47,7 @@ export default class Menu extends PureComponent {
   constructor(props) {
     super(props)
 
+    this.state = {}
```
Comment (bethstevic) [inline packages/react-ui-core/src/Menu/Menu.js]: Why are you setting state as an empty object here and assigning it a value right below?

File: packages/react-ui-core/src/Menu/Menu.js [hunk 3]
```diff
@@ -93,28 +95,26 @@ export default class Menu extends PureComponent {
...
   }
 
   get items() {
-    return this.options.map(option => (
-      typeof option === 'object' ? option.label : option
-    ))
+    return this.options.map(option => ((typeof option === 'object' && option.label !== undefined) ? option.label : option))
+  }
+
+  option() {
```
Comment (mneumark) [inline packages/react-ui-core/src/Menu/Menu.js]: What do you think about renaming this to `highlightedOption` to make it more clear? Also I see no reason this shouldn't be a getter.

File: packages/react-ui-core/src/Menu/Menu.js [hunk 4]
```diff
@@ -93,28 +95,26 @@ export default class Menu extends PureComponent {
...
+    return this.options[this.state.highlightIndex || 0]
   }
 
   @autobind
   handleSelection() {
-    const { onItemSelect, options } = this.props
+    const { onItemSelect } = this.props
 
-    const option = options[this.state.highlightIndex]
+    const value = this.option()
```
Comment (mneumark) [inline packages/react-ui-core/src/Menu/Menu.js]: Why did you rename this to value when you went out of your way to rename the function away from value to make it more clear?

File: packages/react-ui-core/src/Menu/__tests__/Menu-test.js [hunk 5]
```diff
@@ -35,6 +35,8 @@ const objectOptionsWithLabel = [
   },
   {
     label: 'Baz label',
+    value: 'Baz Value',
+    disabled: true,
```
Comment (mneumark) [inline packages/react-ui-core/src/Menu/__tests__/Menu-test.js]: Do we need some tests for validating what happens when you disable something?

File: packages/react-ui-core/src/Menu/Menu.js [hunk 6]
```diff
@@ -93,28 +95,26 @@ export default class Menu extends PureComponent {
...
     return this.props.options || []
   }
 
   get items() {
-    return this.options.map(option => (
-      typeof option === 'object' ? option.label : option
-    ))
+    return this.options.map(option => ((typeof option === 'object' && option.label !== undefined) ? option.label : option))
+  }
+
```
Comment (bethstevic) [inline packages/react-ui-core/src/Menu/Menu.js]: Can we break this into multiple lines? Like this:
```
return this.options.map(option => 
       ((typeof option === 'object' && option.label !== undefined) ? 
             option.label :
             option
        )
     )
```

File: packages/react-ui-core/src/Menu/__tests__/Menu-test.js (hunk 5 above)
Comment (tadjohnston) [reply packages/react-ui-core/src/Menu/__tests__/Menu-test.js]: Yes!

File: packages/react-ui-core/src/Menu/Menu.js (hunk 4 above)
Comment (tadjohnston) [reply packages/react-ui-core/src/Menu/Menu.js]: Some of this was me having these renamed. I don't think it was ever finished...

File: packages/react-ui-core/src/Menu/__tests__/Menu-test.js (hunk 5 above)
Comment (yasabere) [reply packages/react-ui-core/src/Menu/__tests__/Menu-test.js]: There is a test for that already but the description might not be clear enough

File: packages/react-ui-core/src/Menu/Menu.js (hunk 3 above)
Comment (yasabere) [reply packages/react-ui-core/src/Menu/Menu.js]: Fixed

File: packages/react-ui-core/src/Menu/Menu.js (hunk 1 above)
Comment (yasabere) [reply packages/react-ui-core/src/Menu/Menu.js]: Fixed

File: packages/react-ui-core/src/Menu/__tests__/Menu-test.js (hunk 5 above)
Comment (yasabere) [reply packages/react-ui-core/src/Menu/__tests__/Menu-test.js]: Fixed

Comment (mneumark) [reply packages/react-ui-core/src/Menu/__tests__/Menu-test.js]: Okay, to be clear, he wrote tests in the PR from 2 days ago but had issues with that resolution and in this PR he is fixing stuff but the tests from before remain.

File: packages/react-ui-core/src/Menu/Menu.js (hunk 4 above)
Comment (nray621) [reply packages/react-ui-core/src/Menu/Menu.js]: I'm still confused why this is `value` rather than `option`. If it is named `value` then we end up with an object named `value` that has a `value` attribute

File: packages/react-ui-core/src/Menu/Menu.js [hunk 7]
```diff
@@ -93,28 +93,31 @@ export default class Menu extends PureComponent {
...
+    return this.options[this.state.highlightIndex || 0]
   }
 
   @autobind
   handleSelection() {
-    const { onItemSelect, options } = this.props
+    const { onItemSelect } = this.props
 
-    const option = options[this.state.highlightIndex]
+    const value = this.highlightedOption
```
Comment (nray621) [inline packages/react-ui-core/src/Menu/Menu.js]: I'm still confused why this is `value` rather than `option`. If it is named `value` then we end up with an object named `value` that has a `value` attribute

**Comment by nray621** [comment]
LGTM so far 👍

**Comment by nray621** [comment]
👍
